### PR TITLE
Read backcompat keys in TableMetadataSet extraction

### PR DESCRIPTION
**Pull request by clairelin135**
## Summary & Motivation
Enable defining a mapping of backcompat keys to current keys in `TableMetadataSet`. 

Used to convert old materializations containing `relation_identifier` into a `TableMetadataSet` object with `table_name` populated.

## How I Tested These Changes
BK

## Changelog
NOCHANGELOG

### CODE REVIEW

**Comment by clairelin135** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25403?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25403** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25403?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25400** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25400?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clairelin135 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/metadata/metadata_set.py [hunk 1]
```diff
@@ -102,6 +110,11 @@ def _extract_value(cls, field_name: str, value: Any) -> Any:
         """Based on type annotation, potentially coerce the value to the expected type."""
         ...
 
+    @classmethod
```
Comment (benpankow) [inline python_modules/dagster/dagster/_core/definitions/metadata/metadata_set.py:113]: I wonder if we'll run into situations in the future where we'll want to do value remapping also (e.g. we end up changing how the value for a specific metadata set is represented)

**Review by benpankow — APPROVED** [review]
🙏

File: python_modules/dagster/dagster/_core/definitions/metadata/metadata_set.py (hunk 1 above)
Comment (clairelin135) [reply python_modules/dagster/dagster/_core/definitions/metadata/metadata_set.py:113]: definitely possible! open to changing the APIs to support that in the future

**Pull request merged**